File: 염성훈/12.Programmers_lv1,lv2/pro_로또의 최고 순위와 최저 순위.py
def check_rank(num):
    if num == 6:
        return 1
    elif num == 5:
        return 2
    elif num == 4:
        return 3
    elif num == 3:
        return 4
    elif num == 2:
        return 5
    else:
        return 6

def solution(lottos, win_nums):
    answer = []
    high = 0
    low = 0
    for lotto in lottos:
        if lotto in win_nums:
            low += 1
        if lotto == 0:
            high += 1
    high += low
    answer.append(check_rank(high))
    answer.append(check_rank(low))

    return answer

File: AHYEON/12.programers_lv1_lv2/pro_로또의최고순위와최저순위.swift
import Foundation


func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {
    var check: Int = 0
    var unknown: Int = 0

    for lotto in lottos {
        if win_nums.contains(lotto) {
            check += 1
        }
        else if lotto == 0 {
            unknown += 1
        }
    }
    func getRank(_ num: Int) -> Int {
        switch num {
            case 6: return 1
            case 5: return 2
            case 4: return 3
            case 3: return 4
            case 2: return 5
            default: return 6
        }
    }

    return [getRank(check+unknown), getRank(check)]
}

func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {

    let zeroCount = lottos.filter { $0 == 0}.count
    let winCount: Int = win_nums.filter { lottos.contains($0) }.count

    return [min(7-winCount-zeroCount,6), min(7-winCount,6)]
}